ndc-client/src/models.rs

```diff
@@ -150,6 +150,11 @@ pub enum Type {
         /// The type of the elements of the array
         element_type: Box<Type>,
     },
+    /// A predicate type for a given collection
```

Comment:
Should we specify that a value of this type needs to parse into the `Expression` type?

Reply:
Yes, when I write the spec pages, this will be mentioned there.